genanki/card.py

```diff
@@ -1,11 +1,19 @@
+# Using the epoch milliseconds of when the Card was created as ANKI do
+import time
+current_milli_time = lambda: int(round(time.time() * 1000))
+
 class Card:
   def __init__(self, ord, suspend=False):
     self.ord = ord
     self.suspend = suspend
 
   def write_to_db(self, cursor, now_ts, deck_id, note_id):
     queue = -1 if self.suspend else 0
-    cursor.execute('INSERT INTO cards VALUES(null,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?);', (
+    card_id = current_milli_time()
+    # Wait for 1 milliseconds to ensure that id is unique
+    time.sleep(.001)
```

Comment:
This is not a good solution, because it will make genanki slow for large decks.

Why not pass in an additional field, say `card_idx`, that is incremented by 1 for each card that is inserted into the database? Then you can compute `card_id` based on `now_ts` and `card_idx`.

Reply:
You're right , i hadn't thought enought in the way to implement the solution and i knew that it will slow down genanki for large decks . But of course ,it's a good idea to to pass an additional value to write_to_db, say card_idx, that is incremented by 1 for each card that is inserted into the database. Then you can compute card_id based on now_ts and card_idx.
Thank you !
h